### ISSUE TITLE
Improving colour response?

### ISSUE THREAD

**Issue by kubrickfr** [ILiebhardt/darktable#7]
I know it's very subjective but I really feel we are now traveling along the right edge of your "triangle of challenges".
False colour artifacts are long gone, color bleeding got somewhat better but colour response is still very subdued. I see that vey often in small colour patches like flowers

In this example, you can see that the flowers have lost a lot of their vivid red colour and green spaces between them suffer from very bad colour bleeding and they are just pink-ish rather than green.

![20170604_0217_07](https://user-images.githubusercontent.com/2634188/27859352-69a1a35a-6170-11e7-882a-c447d2a1779a.jpg)
![20170604_0217_06](https://user-images.githubusercontent.com/2634188/27859353-69bcadbc-6170-11e7-8dc3-3c3fe8d86448.jpg)
_comparison between FDC (top) and Markesteijn (bottom)_

Can anything be done to improve that?

**Comment by ILiebhardt** [comment]
If it's about subdued color response, I would start trying to remove one more median filter. I already went from three to two because I found the colors too subdued, and I would try to remove one more. The following steps would be to also address the initial convolution filters, but then you would really risk more color artifacts again....

**Comment by kubrickfr** [comment]
I've removed the median filtering completely before and found that it made no visible difference in most cases.
Have you considered doing Non-uniform discrete Fourier transforms rather than doing a first interpolation step? Just assuming here that the poor colour response _could_ be due partly to the interpolation "softening" the edges...

**Comment by ILiebhardt** [comment]
I haven't considered using a non-uniform discrete Fourier transform yet (but I'm open for suggestions ;-)).
However, I'm quite confident that I can fix this just by making narrower filters. What happened is that I initially made rather broad filters, then I noticed that two of the previously six initial filters are just linear combinations of other values, so I went from six down to four initial filters, to save processing time and memory. Doing this also slightly impacted the color response, so I just need to adapt the filters slightly. My other machine which has the code for doing that is in repair at the moment, but I can do this once it is back - no problem. My main priorities now are the weird green border and cleaning up the still experimental parts of the code.
